Clockwork/Support/Mezzio/config.php

```diff
@@ -0,0 +1,278 @@
+
+		// Sample the collected requests (eg. set to 100 to collect only 1 in 100 requests)
+		'sample' => getenv('CLOCKWORK_REQUESTS_SAMPLE') !== false ? getenv('CLOCKWORK_REQUESTS_SAMPLE') : false,
+
+		// List of URIs that should not be collected
+		'except' => [
+			// '/api/.*'
+		],
+
+		// List of URIs that should be collected, any other URI will not be collected if not empty
+		'only' => [
+			// '/api/.*'
+		],
+
+		// Don't collect OPTIONS requests, mostly used in the CSRF pre-flight requests and are rarely of interest
+		'except_preflight' => getenv('CLOCKWORK_REQUESTS_EXCEPT_PREFLIGHT') !== false ? getenv('CLOCKWORK_REQUESTS_EXCEPT_PREFLIGHT') : true
+	],
+
+	/*
+	|------------------------------------------------------------------------------------------------------------------
+	| Enable data collection when Clockwork is disabled
+	|------------------------------------------------------------------------------------------------------------------
+	|
+	| You can enable this setting to collect data even when Clockwork is disabled. Eg. for future analysis.
+	|
+	*/
+
+	'collect_data_always' => getenv('CLOCKWORK_COLLECT_DATA_ALWAYS') !== false ? getenv('CLOCKWORK_COLLECT_DATA_ALWAYS') : false,
+
+	/*
+	|------------------------------------------------------------------------------------------------------------------
+	| Clockwork API URI
+	|------------------------------------------------------------------------------------------------------------------
+	|
+	| Path of the script calling returnRequest to return Clockwork metadata to the client app. See installation
+	| instructions for details.
+	|
+	*/
+
+	'api' => getenv('CLOCKWORK_API') !== false ? getenv('CLOCKWORK_API') : '/__clockwork/',
+
+	/*
+	|------------------------------------------------------------------------------------------------------------------
+	| Clockwork web UI
+	|------------------------------------------------------------------------------------------------------------------
+	|
+	| Clockwork comes bundled with a full Clockwork App accessible as a Web UI. Here you can enable and configure this
+	| feature.
+	| Clockwork::returnWeb api is used to expose the Web UI in your vanilla app, see the installation instructions for
+	| details.
+	|
+	*/
+
+	'web' => [
+		// Enable or disable the Web UI, set to the public uri where Clockwork Web UI is accessible
+		'enable' => getenv('CLOCKWORK_WEB_ENABLE') !== false ? getenv('CLOCKWORK_WEB_ENABLE') : true,
+
+		// Path where to install the Web UI assets, should be publicly accessible
+		'path' => getenv('CLOCKWORK_WEB_PATH') !== false ? getenv('CLOCKWORK_WEB_PATH') : 'public/vendor/clockwork',
+
+		// Public URI where the installed Web UI assets will be accessible
+		'uri' => getenv('CLOCKWORK_WEB_URI') !== false ? getenv('CLOCKWORK_WEB_URI') : '/vendor/clockwork',
+
+		// host where web clockwork is accessible, used when a server side request return the html
```

Comment:
I believe this is what `CLOCKWORK_WEB_URI` is supposed to be.

Reply:
As explained by the documentation to serve the WEB UI you need to specify this routing:

Reply:
The Web UI in the Vanilla integration works in a bit of a funny way. It actually copies the Web UI implementation to a public directory, so the assets are served directly by the web server, this is by default `public/vendor/clockwork`. The code handling the main web UI route, typically `/clockwork`, then returns a special full-screen iframe that embeds the Web UI located at `/vendor/clockwork` with some special configuration to make it work.

Reply:
This is something that is not really clear and I believe that the different integrations differ a bit, too. Just for reference, here are the two settings I found:
```php
		// Path where to install the Web UI assets, should be publicly accessible
		'path' => isset($_ENV['CLOCKWORK_WEB_PATH']) ? $_ENV['CLOCKWORK_WEB_PATH'] : __DIR__ . '/../../../../../public/vendor/clockwork',

		// Public URI where the installed Web UI assets will be accessible
		'uri' => isset($_ENV['CLOCKWORK_WEB_URI']) ? $_ENV['CLOCKWORK_WEB_URI'] : '/vendor/clockwork'
```
Note that both are commented with the words "accessible" and "public" and both contain a path, which is a bit confusing.

Anyhow, just out of interest, why does the web UI copy the files between paths? I've use the `Web` class' assets and served them from where they are for my PSR-based middleware implementation.

Reply:
@itsgoingd 

> The Web UI in the Vanilla integration works in a bit of a funny way. It actually copies the Web UI implementation to a public directory, so the assets are served directly by the web server, this is by default `public/vendor/clockwork`. The code handling the main web UI route, typically `/clockwork`, then returns a special full-screen iframe that embeds the Web UI located at `/vendor/clockwork` with some special configuration to make it work.

This is exactly how I understood when using it. I only added a new env vars `WEB_HOST` to configure the `/clockwork` url but if you think it's useless I can hadcode the url in the middleware and remove it.